script/tool/lib/src/update_excerpts_command.dart

```diff
@@ -23,232 +26,302 @@ class UpdateExcerptsCommand extends PackageLoopingCommand {
+  static const String _pragma = '<?code-excerpt';
+  static final RegExp _basePattern =
+      RegExp(r'^ *<\?code-excerpt path-base="([^"]+)"\?>$');
+  static final RegExp _injectPattern = RegExp(
+    r'^ *<\?code-excerpt "(?<path>[^ ]+) \((?<section>[^)]+)\)"(?: plaster="(?<plaster>[^"]*)")?\?>$',
+  );
 
-  /// Runs the injection step to update [targetPackage]'s top-level .md files
-  /// with the latest excerpts from [example], returning true on success.
-  Future<bool> _injectSnippets(
-    RepositoryPackage example, {
-    required RepositoryPackage targetPackage,
-  }) async {
-    final List<String> relativeMdPaths = targetPackage.directory
-        .listSync()
-        .whereType<File>()
-        .where((File f) =>
-            f.basename.toLowerCase().endsWith('.md') &&
-            // Exclude CHANGELOG since it should never have excerpts.
-            f.basename != 'CHANGELOG.md')
-        .map((File f) => getRelativePosixPath(f, from: example.directory))
-        .toList();
-    if (relativeMdPaths.isEmpty) {
-      return true;
+  _UpdateResult _updateExcerptsIn(File file) {
+    bool detectedChange = false;
+    List<String>? errors;
+    void reportError(String message) {
+      errors ??= <String>[];
+      errors!.add(message);
     }
-    final int exitCode = await processRunner.runAndStream(
-        'dart',
-        <String>[
-          'run',
-          'code_excerpt_updater',
-          '--write-in-place',
-          '--yaml',
-          '--no-escape-ng-interpolation',
-          ...relativeMdPaths,
-        ],
-        workingDir: example.directory);
-    return exitCode == 0;
-  }
 
-  /// Adds `code_excerpter` and `code_excerpt_updater` to [package]'s
-  /// `dev_dependencies` using path-based references to the submodule copies.
-  ///
-  /// This is done on the fly rather than being checked in so that:
-  /// - Just building examples don't require everyone to check out submodules.
-  /// - Examples can be analyzed/built even on versions of Flutter that these
-  ///   submodules do not support.
-  void _addSubmoduleDependencies(RepositoryPackage package,
-      {required Directory repoRoot}) {
-    final String pubspecContents = package.pubspecFile.readAsStringSync();
-    // Save aside a copy of the current pubspec state. This allows restoration
-    // to the previous state regardless of its git status at the time the script
-    // ran.
-    package.directory
-        .childFile(_originalPubspecFilename)
-        .writeAsStringSync(pubspecContents);
-
-    // Update the actual pubspec.
-    final YamlEditor editablePubspec = YamlEditor(pubspecContents);
-    const String devDependenciesKey = 'dev_dependencies';
-    final YamlNode root = editablePubspec.parseAt(<String>[]);
-    // Ensure that there's a `dev_dependencies` entry to update.
-    if ((root as YamlMap)[devDependenciesKey] == null) {
-      editablePubspec.update(<String>['dev_dependencies'], YamlMap());
+    String pathBase = path.dirname(file.path);
```

Comment:
We generally prefer using the `file` package constructs to working with paths manually. So we would generally do `Directory baseDir = file.directory` and then use `childDirectory`, `childFile`, and/or one of the `child*WithSubcomponents` helpers in `file_utils.dart`.